src/Common/src/CoreLib/System/Threading/Tasks/ConcurrentExclusiveSchedulerPair.cs

```diff
@@ -654,7 +654,7 @@ private bool TryExecuteTaskInlineOnTargetScheduler(Task task)
                 catch
                 {
                     Debug.Assert(t.IsFaulted, "Task should be faulted due to the scheduler faulting it and throwing the exception.");
-                    var ignored = t.Exception;
+                    _ = t.Exception;
```

Comment:
Can you explain why this line is needed at all? I wasn't able to understand the exact side-effect from the docs that makes this necessary (around exception propogation). https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.exception?view=netframework-4.8#remarks

Reply:
Let's say you have:
```C#
Task t = Task.Run(() => { ... });
```
and an exception occurs in that `...`.  We don't just crash as would happen if this were in `ThreadPool.QueueUserWorkItem`, as we want to propagate the exception where you eventually consume the task, e.g.
```C#
await t; // should throw here
```
That means that the Task needs to capture the exception and store it for later, rather than letting it immediately propagate up the stack, go unhandled, and crash. But, what guarantee do we have that you'll actually end up consuming the Task and thus observing its captured exception?  None.  So when Task was first introduced, we added the notion of exception observation, where a faulted Task effectively tracks whether its exception has been observed.  You can "observe" the exception in a variety of ways, such as awaiting the Task (such that the exception is thrown), using .Wait/Result/etc. (such that the exception is thrown), or even just accessing the Task's Exception property, since in that case we're giving you the Exception object and we can be satisfied that you've at least seen it.  If you never observed it, then a finalizer would end up running eventually.  That finalizer would raise the TaskScheduler.UnobservedTaskException event, and then if it was still unobserved, the exception would propagate on the finalizer thread and crash the app.  In .NET 4.5, when async/await were introduced, we backed off this a bit, and made the crashing opt-in rather than opt-out; we still track whether exceptions are observed, and we still invoke the UnobservedTaskException event, but by default unobserved exceptions don't cause a crash, though developers can opt-in to the crashing behavior.

This `_ = t.Exception` is us observing the exception.

Reply:
Thanks for the context and detailed explanation. It makes more sense to me now.